holoviews/util.py

```diff
@@ -47,26 +47,30 @@ class Dynamic(param.ParameterizedFunction):
     def __call__(self, map_obj, **params):
         self.p = param.ParamOverrides(self, params)
         callback = self._dynamic_operation(map_obj)
+        streams = self._get_streams(map_obj)
         if isinstance(map_obj, DynamicMap):
             dmap = map_obj.clone(callback=callback, shared_data=self.p.shared_data,
-                                 streams=[])
+                                 streams=streams)
         else:
-            dmap = self._make_dynamic(map_obj, callback)
-        if isinstance(self.p.operation, ElementOperation):
-            streams = []
-            for stream in self.p.streams:
-                if inspect.isclass(stream) and issubclass(stream, Stream):
-                    stream = stream()
-                elif not isinstance(stream, Stream):
-                    raise ValueError('Stream must only contain Stream '
-                                     'classes or instances')
+            dmap = self._make_dynamic(map_obj, callback, streams)
+        return dmap
+
+
+    def _get_streams(self, map_obj):
+        streams = util.dimensioned_streams(map_obj) if isinstance(map_obj, DynamicMap) else []
+        for stream in self.p.streams:
+            if inspect.isclass(stream) and issubclass(stream, Stream):
+                stream = stream()
```

Comment:
I remember this bit - we are instantiating streams from the classes but I've forgotten why...